### PR TITLE
RCF-214, RCF-212 and RCF-216

### PR DESCRIPTION
**Pull request by SachinPremkumar**
[RCF-214], [RCF-212], [RCF-260] and [RCF-216]

[RCF-214]: https://mosip.atlassian.net/browse/RCF-214?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RCF-212]: https://mosip.atlassian.net/browse/RCF-212?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RCF-216]: https://mosip.atlassian.net/browse/RCF-216?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[RCF-260]: https://mosip.atlassian.net/browse/RCF-260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                String key = it.next();
+                ageGroups.put(key, ageGroupConfig.getString(key));
+            }
+
+            Map context = new HashMap();
+            MVEL.eval(getScript(scriptName), context);
+            context.put("identity", dataContext);
+            context.put("ageGroups", ageGroups);
+            return MVEL.eval("return getApplicantType();", context, String.class);
+        } catch (Throwable t) {
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java:110]: we do not catch Throwable object

File: android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            Log.e(getClass().getSimpleName(),"Failed to evaluate mvel script", t);
+        }
+        return null;
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java:113]: returning null and having no check where it's called is error prone

File: android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+    private String getScript(String scriptName) {
+        if(SCRIPT_CACHE.containsKey(scriptName) && SCRIPT_CACHE.get(scriptName) != null)
+            return SCRIPT_CACHE.get(scriptName);
+
+        try {
+            Optional<FileSignature> fileSignature = this.fileSignatureRepository.findByFileName(scriptName);
+            if(!fileSignature.isPresent()) {
+                Log.e("File signature not found : {}", scriptName);
+                return null;
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java:124]: returning null and and not handling where it's being called is error prone

File: android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                bytes = FileUtils.readFileToByteArray(path.toFile());
+            }
+            String actualData = String.format("{\"hash\":\"%s\"}", HMACUtils2.digestAsPlainText(bytes));
+            if(!validateScriptSignature(fileSignature.get().getSignature(), actualData)) {
+                Log.e("File signature validation failed : {}", scriptName);
+                return null;
+            }
+            SCRIPT_CACHE.put(scriptName, new String(bytes));
+
+        } catch (Throwable t) {
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java:144]: avoid catching throwable object

File: android/app/src/main/java/io/mosip/registration_client/api_services/MasterDataSyncApi.java [hunk 5]
```diff
@@ -142,27 +149,28 @@ public MasterDataSyncApi(ClientCryptoManagerService clientCryptoManagerService,
         this.languageRepository = languageRepository;
         this.jobManagerService = jobManagerService;
         this.auditManagerService = auditManagerService;
+        this.masterDataService = masterDataService;
     }
 
 
-    private void syncPolicyKey(@NonNull MasterDataSyncPigeon.Result<MasterDataSyncPigeon.Sync> result) {
+    private void syncPolicyKey(@NonNull MasterDataSyncPigeon.Result<MasterDataSyncPigeon.Sync> result, @NonNull String APP_ID, @NonNull String REF_ID, @NonNull String SET_APP_ID, @NonNull String SET_REF_ID) {
```
Comment (jainhitesh9998) [inline android/app/src/main/java/io/mosip/registration_client/api_services/MasterDataSyncApi.java:156]: variable name within the function shouldn't be in this format
"REF_ID", it's only used for constants.

what does REF_ID and SET_APP_ID SET_REF_ID and SET_REF_ID mean here, this should be explained

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/MasterDataServiceImpl.java [hunk 6]
```diff
@@ -196,25 +206,25 @@ public void manualSync() {
     @Override
     public void syncCertificate() {
         syncCertificate(() -> {
-        });
+        },"","","","");
     }
 
-    private void syncCertificate(Runnable onFinish) {
+    private void syncCertificate(Runnable onFinish,String APP_ID, String REF_ID, String SET_APP_ID, String SET_REF_ID) {
```
Comment (jainhitesh9998) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/MasterDataServiceImpl.java:212]: can we rename the parameters here , they are confusing

**Review by jainhitesh9998 — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java (hunk 1 above)
Comment (SachinPremkumar) [reply android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java]: I have made changes @jainhitesh9998

File: android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java (hunk 2 above)
Comment (SachinPremkumar) [reply android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java:113]: we are already handling null exception in the usage methods @jainhitesh9998

File: android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java (hunk 3 above)
Comment (SachinPremkumar) [reply android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java:124]: I connected with Himaja. if file signature is null while evaluating Mvel Script we are handling it in usage @jainhitesh9998

File: android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java (hunk 4 above)
Comment (SachinPremkumar) [reply android/app/src/main/java/io/mosip/registration_client/api_services/DocumentCategoryApi.java]: I have made changes @jainhitesh9998

File: android/app/src/main/java/io/mosip/registration_client/api_services/MasterDataSyncApi.java (hunk 5 above)
Comment (SachinPremkumar) [reply android/app/src/main/java/io/mosip/registration_client/api_services/MasterDataSyncApi.java]: I have made changes

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/MasterDataServiceImpl.java (hunk 6 above)
Comment (SachinPremkumar) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/MasterDataServiceImpl.java]: I have made changes

**Review by jainhitesh9998 — APPROVED** [review]

**Pull request merged**